MvvmCross/ViewModels/MvxAppStart.cs

```diff
@@ -57,30 +68,50 @@ protected virtual void Reset()
     public class MvxAppStart<TViewModel> : MvxAppStart
         where TViewModel : IMvxViewModel
     {
-        protected readonly IMvxNavigationService NavigationService;
-
-        public MvxAppStart(IMvxApplication application, IMvxNavigationService navigationService) : base(application)
+        public MvxAppStart(IMvxApplication application, IMvxNavigationService navigationService) : base(application, navigationService)
         {
-            NavigationService = navigationService;
         }
 
-        protected override void Startup(object hint = null)
+        protected override void NavigateToFirstViewModel(object hint)
         {
-            base.Startup(hint);
+            try
+            {
+                NavigationService.Navigate<TViewModel>().GetAwaiter().GetResult();
+            }
+            catch (System.Exception exception)
+            {
+                throw exception.MvxWrap("Problem navigating to ViewModel {0}", typeof(TViewModel).Name);
+            }
+        }
+    }
 
-            NavigateToFirstViewModel(hint);
+    public class MvxAppStart<TViewModel, TParameter> : MvxAppStart<TViewModel> where TViewModel : IMvxViewModel<TParameter>
+    {
+        public MvxAppStart(IMvxApplication application, IMvxNavigationService navigationService) : base(application, navigationService)
+        {
         }
 
-        protected virtual void NavigateToFirstViewModel(object hint)
+        protected override void ApplicationStartup(object hint = null)
         {
-            if (hint != null)
+            base.ApplicationStartup(hint);
+            if (hint is TParameter parameter && Application is IMvxApplication<TParameter> typedApplication)
             {
-                MvxLog.Instance.Trace("Hint ignored in default MvxAppStart");
-            }
+                //There is no way to pass the hint back
+                var typedHint = typedApplication.Startup(parameter);
```

Comment:
What's the point of doing this since we do nothing with the result?  Should it not be above the base call and pass the hint to the `base.ApplicationStartup` instead?